src/main/java/ServiceLayer/UserService.java

```diff
@@ -88,20 +92,27 @@ public Response register(String token, String userName, String password, String
     }
 
     // TODO: add documentation
-    public Response purchaseCart(String token, List<Integer> busketsToBuy, PaymentMethod paymentMethod, SupplyMethod shippingMethod) {
+    public Response purchaseCart(String token, List<Integer> busketsToBuy, String cardNumber, String address) {
         Response response = new Response();
         try {
-            String username = tokenService.extractUsername(token);
-            if (userController.isUserNameExists(username)) {
-                userController.getUserByUsername(username).purchaseCart(busketsToBuy, paymentMethod, shippingMethod);
-                logger.info("User purchase cart successfully: " + username);
-                response.setReturnValue("Purchase cart Succeed");
+            if (_tokenService.validateToken(token)) {
```

Comment:
think about:

if (!_tokenService.validateToken(token)){
                response.setErrorMessage("A user with the username given in the token does not exist.");
                throw new Exception("Invalid session token.");
            }
 if.......